Automattic/wp-calypso
# Implement reusable "Add new site" component

<img width="1728" alt="Screenshot 2024-12-18 at 11 42 11 AM" src="https://github.com/user-attachments/assets/03314bc0-ab59-4427-88b9-d1fb11509093" />




Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/1574

## Proposed Changes

This PR implements a reusable "Add new site" component, which includes:

- All the components required.
- Context API to manage the toggle states for the modals.
- A4A menu items for "Add new site" flow.
- Async loading of env specific data

Note: 

- These changes are behind a feature flag.
- This PR enables this only on the dev & horizon environments. We will enable it on staging & production if everything looks good. 
- This PR implements the flow only on A4A. WPCOM changes will be made in a different PR.
- There are two flows where this was used, on the Overview and Sites dashboard. We have updated it in both places.
- I have kept the icons(WordPress & Pressable) the same as before to be consistent throughout the product. 

Fixing an issue I found here: https://github.com/Automattic/wp-calypso/pull/97636

## Why are these changes being made?

* To create a consistent experience of the "Add new site" flow

## Testing Instructions



* Open the A4A live link below
* Go to the Overview page > Click the `Add sites` button on the top right > Verify the UI looks good on all the devices. Perform all the actions that are available and verify they work as expected. Best thing to do here is compare it with the production version of it by visiting https://agencies.automattic.com/

## Pre-merge Checklist
